src/main/java/com/gregtechceu/gtceu/common/item/DataItemBehavior.java

```diff
@@ -83,28 +80,23 @@ public void appendHoverText(ItemStack stack, @Nullable Level level, List<Compone
     public InteractionResult useOn(UseOnContext context) {
         Level level = context.getLevel();
         BlockPos pos = context.getClickedPos();
-        BlockEntity blockEntity = level.getBlockEntity(pos);
         ItemStack stack = context.getItemInHand();
-        if (level.isClientSide()) {
-            return InteractionResult.PASS;
-        }
-        if (blockEntity instanceof IMachineBlockEntity machineBlockEntity) {
-            MetaMachine machine = machineBlockEntity.getMetaMachine();
-            Pair<GTRecipeType, String> researchData = ResearchManager.readResearchId(stack);
```

Comment:
Do not remove this, data sticks shouldn't have research data + linking data on them.